File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int n = s.size();
        int left = 0;
        int right = 0;
        for(int i = 0 ; i < n ; i++){
            if(s[i] == '(') right++;
            else if(right > 0) right--;
            else left++;
        }
        return left + right;
    }
};

File: 0957-minimum-add-to-make-parentheses-valid/0957-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int ans = 0;
        int l = 0 , r = 0;
        for(char c : s.toCharArray()){
            if(c == '(') l++;
            else if(l > 0) l--;
            else r++;
        }
        return l + r;
    }
}